codegenerator/cli/templates/static/codegen/src/LoadLayer.res

```diff
@@ -196,22 +203,30 @@ let schedule = async loadLayer => {
     // some context.<entitty>.get which are already in the in memory store.
     // This way we'd be able to register more actions,
     // but assuming it would not affect performance in a positive way.
-    // On the other hand it is more predictable, easier for testing and less memory intensive.
-    let _ = await %raw(`void 0`)
+    // On the other hand `await Promise.resolve()` is more predictable, easier for testing and less memory intensive.
+    await Promise.resolve()
 
     let loadActionMaps = loadLayer.byEntity->Js.Dict.values
 
     // Reset loadActionMaps, so they can be filled with new loaders.
     // But don't reset the schedule function, so that it doesn't run before execute finishes.
     loadLayer.byEntity = Js.Dict.empty()
 
+    let promises = []
+    loadActionMaps->Array.forEach(loadActionMap => {
+      switch loadActionMap->LoadActionMap.getIdsToLoad {
+        | [] => ()
+        | idsToLoad => 
+          promises->Js.Array2.push(loadLayer->executeLoadEntitiesById(~idsToLoad, ~loadActionMap))->ignore
+      }
+      switch loadActionMap->LoadActionMap.getIndexes {
+        | [] => ()
+        | lookupByIndex => 
+          promises->Js.Array2.push(loadLayer->executeLoadEntitiesByIndex(~lookupByIndex, ~loadActionMap))->ignore
+      }
```

Comment:
I guess the issue with doing this in parallel is that you can't benefit from values loaded in on get by index on get by id.

If a value that is also fetched by id happens to exist in the where query result then you still fetch it twice.

Since we're also limiting pg connections it may not actually be parallelized.

Also it's important that all the indices are set in the inmemory store before any other values are added so that they can add their ids to each index if they meet the equality check.

Reply:
Hmmm I feel like there are a bunch of things I need to double check with these index queries. I'm not sure if it's resistant to a value already being loaded in to the in memory store before it creates the index 🤔